src/v2/Components/v2/ArtworkFilter/ArtworkFilterMobileActionSheet.tsx

```diff
@@ -75,14 +103,36 @@ export const ArtworkFilterMobileActionSheet: SFC<{
       </Content>
 
       <Footer p={1}>
-        <Button variant="primaryBlack" width="100%" onClick={onClose}>
-          Apply
-        </Button>
+        <ApplyButton
+          onClick={() => {
+            // On apply, replace the actual filter state with the
+            // hitherto staged filters
+            filterContext.setFilters(filterContext.stagedFilters)
+            onClose()
+          }}
+          activeFilterCount={countActiveFilters(filterContext.stagedFilters)}
+        />
       </Footer>
     </ModalBase>
   )
 }
 
+interface ApplyButtonProps {
+  activeFilterCount: number
+  onClick: () => void
+}
+
+const ApplyButton: React.SFC<ApplyButtonProps> = ({
+  activeFilterCount,
+  onClick,
+}) => {
+  return (
+    <Button variant="primaryBlack" width="100%" onClick={onClick}>
+      Apply ({activeFilterCount})
```

Comment:
The result of counting up the pending filters.

Hmm, maybe "pending filter count" would have been a better name? I mean it's a mix of active and pending, so naming is hard here.

Reply:
Should this be called `stagedFilters`? On iOS the logic was to just show the number of filters that hadn't yet been applied here.

Reply:
OH! I didn't realize that about the filter count on iOS. Let's ticket that as a follow-up?